### PR TITLE
add various checks to improve code quality. Fix issues highlighted

### PR DESCRIPTION
**Pull request by maleck13**

### CODE REVIEW

File: pkg/httpclient/externalRequestClient.go [hunk 1]
```diff
@@ -38,3 +40,14 @@ func (hcb *ClientBuilder) Build() mobile.ExternalHTTPRequester {
 	client.Timeout = time.Second * time.Duration(hcb.timeout)
 	return client
 }
+
+func CloseResponseBody(res *http.Response, logger *logrus.Logger) {
```
Comment (philbrookes) [inline pkg/httpclient/externalRequestClient.go]: is this passing the new go-fmt test? It's missing a comment anyway

File: pkg/k8s/serviceCatalog.go [hunk 2]
```diff
@@ -222,7 +226,11 @@ func (sc *serviceCatalogClient) BindToService(bindableService, targetSvcName str
 	if err != nil {
 		return errors.Wrap(err, "failed to create binding. Request error occurred")
 	}
-	defer res.Body.Close()
+	defer func() {
```
Comment (philbrookes) [inline pkg/k8s/serviceCatalog.go]: Why are you doing this as deferred funcs here, and as a deferred call to a closeBody function in the httpClient pkg? I think we should do it the same way - whichever way that is - every where.

File: pkg/httpclient/externalRequestClient.go (hunk 1 above)
Comment (maleck13) [reply pkg/httpclient/externalRequestClient.go]: go fmt doesn't care about comments that is golint

File: pkg/k8s/serviceCatalog.go (hunk 2 above)
Comment (maleck13) [reply pkg/k8s/serviceCatalog.go]: had issues with circular imports. Can change to use the anon func everywhere

File: pkg/httpclient/externalRequestClient.go (hunk 1 above)
Comment (maleck13) [reply pkg/httpclient/externalRequestClient.go]: Meant to remove that function actually

File: pkg/k8s/serviceCatalog.go (hunk 2 above)
Comment (maleck13) [reply pkg/k8s/serviceCatalog.go]: actually as mentioned the httpclient func isnt actually used. Have removed it now

**Pull request merged**